src/microbe_stage/MicrobeStage.cs

```diff
@@ -893,32 +899,64 @@ private void UpdateLinePlayerPosition()
     {
         if (Player == null || Player?.Dead == true)
         {
-            foreach (var chemoreceptionLine in chemoreceptionLines)
-                chemoreceptionLine.Visible = false;
+            foreach (var chemoreception in chemoreceptionLines)
+                chemoreception.Line.Visible = false;
 
             return;
         }
 
-        var position = Player!.GlobalTransform.origin;
+        var position = Player!.GlobalTranslation;
 
-        foreach (var chemoreceptionLine in chemoreceptionLines)
+        foreach (var chemoreception in chemoreceptionLines)
         {
-            if (chemoreceptionLine.Visible)
-                chemoreceptionLine.LineStart = position;
+            if (!chemoreception.Line.Visible)
+                continue;
+
+            chemoreception.Line.LineStart = position;
+
+            // The target needs to be updated for detected microbes but not compounds.
+            if (chemoreception.TargetMicrobe != null)
+            {
+                // Use colony parent position to avoid calling GlobalTranslation
+                if (chemoreception.TargetMicrobe.ColonyParent != null)
```

Comment:
This is not accurate. ColonyParent technically isn't required t obe the colony lead cell. As you actually want the colony lead cell you need to go through the MicrobeColony object and grab the master from there.